tests/resources/generate-outputs.sh

```diff
@@ -7,9 +7,42 @@ ROOT_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
 
 cd "$ROOT_DIR"
 
-mtt train options.yaml -o model-32-bit.pt -r base_precision=32
-mtt train options.yaml -o model-64-bit.pt -r base_precision=64
-mtt train options-pet.yaml -o model-pet.pt
+HASH_FILE=".data_version.txt"
+FORCE_REGENERATE=false
+if [[ "${FORCE_REGENERATE:-0}" == "1" ]]; then
+  echo "FORCE_REGENERATE=1 detected. Forcing regeneration of all models."
+  FORCE_REGENERATE=true
+else
+  if [ -f "$HASH_FILE" ]; then
+    SAVED_HASH=$(cat "$HASH_FILE")
+    CURRENT_HASH=$(git rev-parse HEAD)
+    if [ "$SAVED_HASH" != "$CURRENT_HASH" ]; then
+      echo "Git commit has changed. Forcing regeneration of all models."
+      FORCE_REGENERATE=true
+    fi
+  else
+    echo "Hash file not found. Forcing regeneration of all models."
+    FORCE_REGENERATE=true
+  fi
+fi
+
+# Regenerate if --force is used OR if the file doesn't exist
+if [ "$FORCE_REGENERATE" = true ] || [ ! -f "model-32-bit.pt" ]; then
+    mtt train options.yaml -o model-32-bit.pt -r base_precision=32
+fi
+
+if [ "$FORCE_REGENERATE" = true ] || [ ! -f "model-64-bit.pt" ]; then
+    mtt train options.yaml -o model-64-bit.pt -r base_precision=64
+fi
+
+if [ "$FORCE_REGENERATE" = true ] || [ ! -f "model-pet.pt" ]; then
+    mtt train options-pet.yaml -o model-pet.pt
+fi
+
+if [ "$FORCE_REGENERATE" = true ]; then
+  echo "Saving current git commit hash to version the data."
+  git rev-parse HEAD > "$HASH_FILE"
+fi
```

Comment:
Maybe we loop this? If we add more files.

BTW: we also have some files to generate for the docs 🙈

Reply:
Yes, I'm going to push a bit for making this opt-in for now. This way we don't break anything in the default case

Reply:
> Yes, I'm going to push a bit for making this opt-in for now. This way we don't break anything in the default case

Made an opt-in variant.